fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/FedoraResourceImpl.java

```diff
@@ -563,11 +569,31 @@ public void replaceProperties(final IdentifierConverter<Resource, FedoraResource
             exceptions.append(e.getMessage());
         }
 
+        removeEmptyFragments();
+
         if (exceptions.length() > 0) {
             throw new MalformedRdfException(exceptions.toString());
         }
     }
 
+    private void removeEmptyFragments() {
+        try {
+            if (node.hasNode("#")) {
+                for (final Iterator<Node> hashNodes = node.getNode("#").getNodes(); hashNodes.hasNext(); ) {
+                    final Node n = hashNodes.next();
+                    final Iterator userProps = Iterators.filter(n.getProperties(),
+                            p -> !isManagedPredicate.test(propertyConverter.convert((Property)p)));
+                    if ( !userProps.hasNext() ) {
+                        LOGGER.debug("Removing empty hash URI node: {}", n.getName());
+                        n.remove();
+                    }
+                }
+            }
+        } catch (RepositoryException ex) {
+            LOGGER.warn("Error removing empty fragments: {}", ex.toString());
```

Comment:
We can throw the exception, which would cause the update being processed to be abandoned.  As long as there aren't any errors in our node navigation, then I would expect the errors to be serious enough that abandoning the update probably makes sense.